### ISSUE TITLE
Real measurements.

### ISSUE THREAD

**Issue by callahanp** [Juanvvc/c172p-detailed#1090]
I'm building a cockpit for the C172P.  One problem is the need to get accurate measurements of the actual position and size of various parts of the cockpit, the position and size of various controls, gauges, instruments and radios.  

I'd like to be able to produce accurate blueprint drawings from the flightgear aircraft data.

Ideally, what I'd like to be able to do is sit inside a real one for a few hours with calipers, rulers and a camera, but don't know any pilots well enough to ask for such a favor.  Failing that, I'm wondering if would be possible to add movable rulers and calipers to theflightgear  c172p model as movable instruments.

I'm going to generalize this question for the flightgear dev mailing list, but feel free to comment here on a  specific approaches to this problem for the c172p.

![measurements1](https://user-images.githubusercontent.com/6132176/39772274-cef96884-52c2-11e8-9e84-04d6c96e78f6.jpg)

**Comment by wlbragg** [comment]
Interesting, possible yes, amount of work involved to do that though may be prohibitive. But it would be a really handy tool to have for developing as well.
One question though, you trust then that our model is accurate enough for your purposes?

**Comment by gilbertohasnofb** [comment]
@tigert once sent us a bunch of photos that we used in this project in one way or another. Those might be helpful to you: https://www.dropbox.com/s/5bc75x87l8ezan2/c172p-interior-photos.zip?dl=0

**Comment by callahanp** [comment]
While the current model looks to me to be very good, it has a few areas I'd question:
  - Lower Panel cutout under the radio stack is missing.
  - Some lower panel elements might not be in exactly the right place
  - Instrument Panel vertical distances between the rows of instruments might differ from the real plane
  - The Fuel and Oil Gagues look different from pictures of the actual plane.  Their aspect ratio seems a bit off.

Not having a real plane handy, I can't verify or improve on any of this..

Tigert's photo's were helpful, but the measurement quality varies all over the place due to parallax or placement of the tape in such a way that its twisted, not parallel to an edge or distance from the objects to be measured.  Also a photo is not the best way of obtaining accurate measurements.  

I'm really interested in doing some more accurate measurements of just about every aspect of the cockpit,  There's about 100 objects in there, each of which may need one or more dimensions measured.  I'd use more accurate measurement tools than just a tape measure.  

I'm willing to travel a bit if we can find a flightgear user who is also a 1982 C172P, owner and pilot willing to work along side me on this.  Doing it all might take several hours but I'd settle for not doing every possible measurement if time's a problem.   That's a pretty tall order, but if the pilot also want to fly that day, I'll buy the $100 hamburgers. (burgers, soft drinks, avgas, landing fees, and any other expenses of the day, on me)

So any takers?  

Any Ideas on spreading the word to possibly interested pilots?

-Pat

**Comment by legoboyvdlp** [comment]
Isn't @juanvvc a pilot? I believe he is in Spain. Not entirely sure what model of Cessna he flies / flew. And there is of course OH CTL itself, perhaps tigert might be able to contact the current operators in order to arrange a visit.

**Comment by gilbertohasnofb** [comment]
> Not having a real plane handy, I can't verify or improve on any of this..

This is the relevant bit. If you get real measurements than we can shift things around. Also, there is also a limit of what we want to achieve here, I frankly would not bother moving things around if they are just a few millimeters off their correct positions. It's a matter of pain vs gain, as always. And if no one can tell the difference between our switch positions without a ruler then perhaps we are fine with our current model.

**Comment by gilbertohasnofb** [comment]
> I'm willing to travel a bit if we can find a flightgear user who is also a 1982 C172P, owner and pilot willing to work along side me on this. Doing it all might take several hours but I'd settle for not doing every possible measurement if time's a problem. That's a pretty tall order, but if the pilot also want to fly that day, I'll buy the $100 hamburgers. (burgers, soft drinks, avgas, landing fees, and any other expenses of the day, on me)

Yeah, @tigert had access to one in Finland, I believe, but he went silent since quite some time, probably busy with RL or just having moving on from FG. Where are you based, Pat? I took a look at the airclubs here around London and it seems they all have much more modern versions of the 172, unfortunately.

**Comment by callahanp** [comment]
Based in USA, Framingham, Ma.  I'm investigating my resources here.  Lots
of upgrading going on around here too I hear.

As for making fg c172p changes based on more accurate measurements, what
gets done would be up to whoever wants to make the change, or not.

I will provide you with a copy of any real, well documented data I can put
together, but I realize you each have your own set of priorities and have
to work on what makes sense to you in the time you have available for
flightgear and the C172P.

-Pat

**Comment by legoboyvdlp** [comment]
Our local club here at Ards only has the SP unfortunately. So no good there either.

**Comment by callahanp** [comment]
I just found out about a $400 device that works with an Ipad to do 3d
scans.  Anyone know of someone with one of these?  It's on my wish list and
I've got lots of questions about it's output format and  how we might use
it.
I wonder if there's an open source version of it.

If we used one in a cockpit, what would we get that we could process into
stuff for an aircraft.

I'm thinking one might develop a standard process to use data from such a
device to produce accurate  features textures and hotspots for an aircraft.
Something analogous to what we're doing with scenery data but for Aircraft.

-Pat

**Comment by gilbertohasnofb** [comment]
@legoboyvdlp Ards as in Ireland?

> As for making fg c172p changes based on more accurate measurements, what gets done would be up to whoever wants to make the change, or not. 
> I will provide you with a copy of any real, well documented data I can put together, but I realize you each have your own set of priorities and have to work on what makes sense to you in the time you have available for flightgear and the C172P.

@callahanp That's really an excellent approach and we would certainly welcome more data which may then be used in future improvements! About the 3D scanning thing, the issue is that, as far as I know is, there is a lot of noise in the scanned data and while the results can be impressive they will be way too unoptimized when comparing with a hand modelled aircraft. Look at the amount of data points of something like this:

![](https://cdn.thingiverse.com/renders/fc/16/a3/7a/fa/IMG_0209_preview_featured.jpg)

I think this is more useful for people rendering things than for using for real-time 3D stuff such as games. But as source of measurements and data this could be really wonderful!

**Comment by legoboyvdlp** [comment]
> @legoboyvdlp Ards as in Ireland?

Yes!

**Comment by gilbertohasnofb** [comment]
Cool, I recently relocated to London 😄

**Comment by callahanp** [comment]
That's a lot of data to try to pull a wireframe from..

On thing that might be better with a scan is the lack of perfection in reality would translate to subtle differences in position of certain objects. For example Circuit breakers or switches might be mounted in a set of holes with enough clearance so the repeated position and vertical alignment is not perfect .In the sim we use a single unvarying distance between repeated objects and absolutely everything lines up properly.  Labels and legends are another area for variation.  Wear and tear might wear off one of the letters so the Radio circuit breakers are labeled as 'adios' with a light smudge replacing the R.

And that scan? It looks very much like my missing shoe.  

Adios,
 Pat

**Comment by gilbertohasnofb** [comment]
Well, please do share whatever data you get your hands into! Also high quality photos under neutral light, close ups of the panel texture, labels, etc. could become very helpful (similarly to those photos tigert had originally shared with us, but with more closeups).

> And that scan? It looks very much like my missing shoe.

🤣 So I suppose it is not missing any longer!
